profiling/src/profile/mod.rs

```diff
@@ -961,12 +1170,17 @@ impl TryFrom<&Profile> for pprof::Profile {
             .iter()
             .map(|(sample, values)| {
                 // Clone the labels, but enrich them with endpoint profiling.
-                let mut labels = sample.labels.clone();
-                if let Some(offset) = sample.local_root_span_id_label_offset {
-                    // Safety: this offset was created internally and isn't be mutated.
-                    let lrsi_label = unsafe { sample.labels.get_unchecked(offset) };
+                let mut labels: Vec<pprof::Label> = profile
+                    .get_labelset(sample.labels)
+                    .iter()
+                    .map(|l| (*profile.get_label(*l)).into())
+                    .collect();
```

Comment:
Maybe not for this PR, but now that we de-duplicate labelsets, could we reuse the labelset conversion for all samples that have the same labels?

(I guess this wouldn't make a difference for timeline so maybe it's not worth it, but I'm curious if it's simple, or if with the Rust lifetime rules we'd have a hard time doing it).

Reply:
I'll look at this for a later PR.